### PR TITLE
fact_diff filter plugin

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### SUMMARY

Fixed: https://github.com/ansible-collections/ansible.utils/issues/78

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by NilashishC — CHANGES_REQUESTED** [review]
Please fix the lint issues before merging this PR.

File: plugins/filter/fact_diff.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                        skip_lines:
+                            description:
+                            - Skip lines matching these regular expressions.
+                            - Matches will be removed prior to the diff.
+                            - If the provided I(before) and I(after) are a string, they will be split.
+                            - Each entry in each list will be cast to a string for the comparison
+                            type: list
+                            elements: str
+
+
```
Comment (rohitthakur2590) [inline plugins/filter/fact_diff.py:59]: are multiple new lines required here ?

File: plugins/filter/fact_diff.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+The fact_diff filter plugin
+"""
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: fact_diff
+    author: Ashwini Mhatre ((@amhatre))
```
Comment (rohitthakur2590) [inline plugins/filter/fact_diff.py:17]: are these multiple braces intended?

File: plugins/filter/fact_diff.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                            - Each entry in each list will be cast to a string for the comparison
+                            type: list
+                            elements: str
+
+
+    notes:
+"""
+
+EXAMPLES = """
+"""
```
Comment (rohitthakur2590) [inline plugins/filter/fact_diff.py:64]: examples are missing

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Review by rohitthakur2590 — APPROVED** [review]

File: plugins/filter/fact_diff.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                    default: ansible.utils.native
+                    type: str
+                vars:
+                    description:
+                    - Parameters passed to the diff plugin.
+                    type: dict
+                    default: {}
+                    suboptions:
+                        skip_lines:
+                            description:
```
Comment (NilashishC) [inline plugins/filter/fact_diff.py:51]: Indentation issue here.

File: plugins/filter/fact_diff.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                            - Skip lines matching these regular expressions.
+                            - Matches will be removed prior to the diff.
+                            - If the provided I(before) and I(after) are a string, they will be split.
+                            - Each entry in each list will be cast to a string for the comparison
+                            type: list
+                            elements: str
+    notes:
```
Comment (NilashishC) [inline plugins/filter/fact_diff.py:58]: Is this required?

File: plugins/filter/fact_diff.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    data = dict(zip(keys, args[1:]))
+    data.update(kwargs)
+    aav = AnsibleArgSpecValidator(data=data, schema=DOCUMENTATION, name="fact_diff")
+    valid, errors, updated_data = aav.validate()
+    if not valid:
+        raise AnsibleFilterError(errors)
+    res = fact_diff(**updated_data)
+    import json
+
+    res = str(res)
```
Comment (NilashishC) [inline plugins/filter/fact_diff.py:194]: Please use to_text() here.

File: plugins/filter/fact_diff.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+    keys = ["before", "after", "plugin"]
+    data = dict(zip(keys, args[1:]))
+    data.update(kwargs)
+    aav = AnsibleArgSpecValidator(data=data, schema=DOCUMENTATION, name="fact_diff")
+    valid, errors, updated_data = aav.validate()
+    if not valid:
+        raise AnsibleFilterError(errors)
+    res = fact_diff(**updated_data)
+    import json
```
Comment (NilashishC) [inline plugins/filter/fact_diff.py:192]: Is this import required?

**Review by NilashishC — APPROVED** [review]

**Pull request merged**